### DISCUSSION TITLE
NEF file (from newer camera model) corrupted image when converted to JPG

### DISCUSSION THREAD

**Discussion by ditaz** [dlemstra/Magick.NET#1287] [Help]
Hello,

I have an issue similar to this one https://www.imagemagick.org/discourse-server/viewtopic.php?t=11183

A NEF file when converted to JPG results to corrupted image.
Please see attached of corrupted image.

![sample](https://user-images.githubusercontent.com/4586075/197953390-d201fe71-8ffa-4530-a894-9fc659a74020.jpg)

The raw NEF file originates from Nikon Z2 with a size of 31.1mb.

The Magick.NET version I'm using is the latest one, 12.2.0.

If latest IM version does not support yet NEF file from latest camera models like Nikon Z2, is there a workaround we can implement to address the issue?

Your response will be greatly appreciated.

**Comment by dlemstra** [comment]
Ca you add the original image? You can split it up in multiple zip files and upload it here.

**Reply by ditaz** [reply]
Hello @dlemstra, thank you for your response. 

Here are the zipped split files (splitted into 27parts, 01-27)

[Nikon Z9-01.zip](https://github.com/dlemstra/Magick.NET/files/10009497/Nikon.Z9-01.zip)
[Nikon Z9-02.zip](https://github.com/dlemstra/Magick.NET/files/10009501/Nikon.Z9-02.zip)
[Nikon Z9-03.zip](https://github.com/dlemstra/Magick.NET/files/10009502/Nikon.Z9-03.zip)
[Nikon Z9-04.zip](https://github.com/dlemstra/Magick.NET/files/10009507/Nikon.Z9-04.zip)
[Nikon Z9-05.zip](https://github.com/dlemstra/Magick.NET/files/10009508/Nikon.Z9-05.zip)
[Nikon Z9-06.zip](https://github.com/dlemstra/Magick.NET/files/10009509/Nikon.Z9-06.zip)
[Nikon Z9-07.zip](https://github.com/dlemstra/Magick.NET/files/10009541/Nikon.Z9-07.zip)
[Nikon Z9-08.zip](https://github.com/dlemstra/Magick.NET/files/10009543/Nikon.Z9-08.zip)
[Nikon Z9-09.zip](https://github.com/dlemstra/Magick.NET/files/10009549/Nikon.Z9-09.zip)
[Nikon Z9-10.zip](https://github.com/dlemstra/Magick.NET/files/10009552/Nikon.Z9-10.zip)
[Nikon Z9-11.zip](https://github.com/dlemstra/Magick.NET/files/10009557/Nikon.Z9-11.zip)
[Nikon Z9-12.zip](https://github.com/dlemstra/Magick.NET/files/10009559/Nikon.Z9-12.zip)
[Nikon Z9-13.zip](https://github.com/dlemstra/Magick.NET/files/10009563/Nikon.Z9-13.zip)
[Nikon Z9-14.zip](https://github.com/dlemstra/Magick.NET/files/10009565/Nikon.Z9-14.zip)
[Nikon Z9-15.zip](https://github.com/dlemstra/Magick.NET/files/10009566/Nikon.Z9-15.zip)
[Nikon Z9-16.zip](https://github.com/dlemstra/Magick.NET/files/10009567/Nikon.Z9-16.zip)
[Nikon Z9-17.zip](https://github.com/dlemstra/Magick.NET/files/10009568/Nikon.Z9-17.zip)
[Nikon Z9-18.zip](https://github.com/dlemstra/Magick.NET/files/10009570/Nikon.Z9-18.zip)
[Nikon Z9-19.zip](https://github.com/dlemstra/Magick.NET/files/10009572/Nikon.Z9-19.zip)
[Nikon Z9-20.zip](https://github.com/dlemstra/Magick.NET/files/10009573/Nikon.Z9-20.zip)
[Nikon Z9-21.zip](https://github.com/dlemstra/Magick.NET/files/10009574/Nikon.Z9-21.zip)
[Nikon Z9-22.zip](https://github.com/dlemstra/Magick.NET/files/10009576/Nikon.Z9-22.zip)
[Nikon Z9-23.zip](https://github.com/dlemstra/Magick.NET/files/10009577/Nikon.Z9-23.zip)
[Nikon Z9-24.zip](https://github.com/dlemstra/Magick.NET/files/10009587/Nikon.Z9-24.zip)
[Nikon Z9-25.zip](https://github.com/dlemstra/Magick.NET/files/10009588/Nikon.Z9-25.zip)
[Nikon Z9-26.zip](https://github.com/dlemstra/Magick.NET/files/10009589/Nikon.Z9-26.zip)
[Nikon Z9-27.zip](https://github.com/dlemstra/Magick.NET/files/10009592/Nikon.Z9-27.zip)

Here is the parent file that needs to extract after unzipping 01-27 part files above
[Nikon Z9.zip](https://github.com/dlemstra/Magick.NET/files/10009595/Nikon.Z9.zip)

Just let me know if encounter any issues opening the file.
Thanks again!

**Reply by ditaz** [reply]
hi @dlemstra , I'd just like to ask if you've had the chance to open the file?

**Reply by dlemstra** [reply]
Sorry but it looks like I forgot to reply that I cannot open that file. Can you use 7zip instead and limit the number of files? The max file size is 25 so you don't need to zip it in individual files of 1mb.

**Reply by ditaz** [reply]
hi @dlemstra thank you for the reply. Sorry but we I'm unable to provide a  NEF file smaller than 25mb. Is it possible to share it through one drive link? 
If not, I will try 7zip and we will see if you can open the file.

**Reply by dlemstra** [reply]
What you can do is split the zip in multiple files but limit each file size to 25mb instead of 1mb. Also feel free to use a OneDrive link and I can then add the file here again for future reference.

**Reply by ditaz** [reply]
hello @dlemstra,
Here's the zip files
[Nikon Z9-01.zip](https://github.com/dlemstra/Magick.NET/files/10810017/Nikon.Z9-01.zip)
[Nikon Z9-02.zip](https://github.com/dlemstra/Magick.NET/files/10810018/Nikon.Z9-02.zip)

here's one drive link in case you cannot open the files above

https://1drv.ms/u/s!ArhBP6E6L1P9gd8WlxmZAzGhWRdVfw?e=NdhaOh

Thank you

**Comment by dlemstra** [comment]
Adding zip files here for future reference:
[Nikon Z9.zip.001.txt](https://github.com/dlemstra/Magick.NET/files/10821553/Nikon.Z9.zip.001.txt)
[Nikon Z9.zip.002.txt](https://github.com/dlemstra/Magick.NET/files/10821557/Nikon.Z9.zip.002.txt)

**Comment by dlemstra** [comment]
I am unable to read this file with the latest version of ImageMagick/Magick.NET. It looks like there is an issue with this file.

**Reply by dlemstra** [reply]
The reason that this file is not working is lack of support for reading this file in the `libraw` library that we are using.
